### PR TITLE
 USS Mayflower

### PR DESCRIPTION
**Pull request by VividPups**
## About the PR
Adds a New Expedition/Bounty hunter ship.  The USS Mayflower 

## Why / Balance
As far as I was able to see the only Bounty hunter/Expedition ship was the KC Sprinter. So I would like to add more that is more dedicated in bounty hunting/ Expeditions

## Media
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added The USS Mayflower

### CODE REVIEW

**Comment by dvir001** [comment]
image
Names need to match, lowercase only

File: Resources/Prototypes/_NF/Shipyard/mayflower.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  id: Mayflower
+  mapName: 'USS Mayflower'
+  mapPath: /Maps/Shuttles/Mushroom.yml
+  minPlayers: 0
+  stations: 
+    Mayflower:
+      stationProto: StandardFrontierExpeditionVessel
+      components:
+        - type: StationNameSetup
+          mapNameTemplate: 'USS Mayflower{1}'
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/mayflower.yml:20]: Name format missing space

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 2]
```diff
@@ -0,0 +1,5827 @@
...
+      parent: 1
+      type: Transform
+- proto: KitchenMicrowave
+  entities:
+  - uid: 578
+    components:
+    - pos: -3.5,-5.5
+      parent: 1
+      type: Transform
+- proto: LockerCaptain
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:3761]: Please keep it to base line lockers, pre filled.

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 3]
```diff
@@ -0,0 +1,5827 @@
...
+  - uid: 394
+    components:
+    - flags: InContainer
+      type: MetaData
+    - parent: 393
+      type: Transform
+    - canCollide: False
+      type: Physics
+    - type: InsideEntityStorage
+- proto: ClothingOuterCoatHoSTrench
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:2359]: Hos gear?? this is sec gear.

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 4]
```diff
@@ -0,0 +1,5827 @@
...
+            - [color=yellow]Piercing[/color] damage reduced by [color=lightblue]10%[/color].
+
+            - [color=yellow]Caustic[/color] damage reduced by [color=lightblue]10%[/color].
+
+            - [color=orange]Explosion[/color] damage reduced by [color=lightblue]10%[/color].
+          priority: 0
+          component: Armor
+        title: null
+      type: GroupExamine
+- proto: ClothingOuterStraightjacket
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:2402]: Not sure if we want this mapped ever.

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 5]
```diff
@@ -0,0 +1,5827 @@
...
+      type: Physics
+    - type: InsideEntityStorage
+- proto: ClothingShoesBootsCombatFilled
+  entities:
+  - uid: 863
+    components:
+    - pos: -5.5,-1.5
+      parent: 1
+      type: Transform
+- proto: ClothingUniformJumpskirtDetective
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:2420]: Detective gear?

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 6]
```diff
@@ -0,0 +1,5827 @@
...
+  - uid: 396
+    components:
+    - flags: InContainer
+      type: MetaData
+    - parent: 393
+      type: Transform
+    - canCollide: False
+      type: Physics
+    - type: InsideEntityStorage
+- proto: ClothingUniformJumpsuitDetective
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:2438]: Detective gear?

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 7]
```diff
@@ -0,0 +1,5827 @@
...
+  - uid: 478
+    components:
+    - rot: 1.5707963267948966 rad
+      pos: 4.5,-5.5
+      parent: 1
+      type: Transform
+    - ShutdownSubscribers:
+      - 460
+      type: DeviceNetwork
+- proto: FloorDrain
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:3000]: Drain in a merc and medical ship?

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 8]
```diff
@@ -0,0 +1,5827 @@
...
+    - isPlaceable: True
+      type: PlaceableSurface
+- proto: LockerParamedicFilledHardsuit
+  entities:
+  - uid: 579
+    components:
+    - pos: 0.5,-5.5
+      parent: 1
+      type: Transform
+- proto: LockerSalvageSpecialistFilledHardsuit
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:3820]: Salvage locker also?

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 9]
```diff
@@ -0,0 +1,5827 @@
...
+      type: Transform
+- proto: WarpPointShip
+  entities:
+  - uid: 868
+    components:
+    - rot: 1.5707963267948966 rad
+      pos: -0.5,-6.5
+      parent: 1
+      type: Transform
+- proto: WeaponRevolverMateba
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:5748]: Detective gun?

File: Resources/Maps/Shuttles/Mushroom.yml [hunk 10]
```diff
@@ -0,0 +1,5827 @@
...
+  entities:
+  - uid: 851
+    components:
+    - pos: -5.4586143,-1.5184155
+      parent: 1
+      type: Transform
+- proto: WeaponShotgunSawn
```
Comment (dvir001) [inline Resources/Maps/Shuttles/Mushroom.yml:5755]: Pre cut shotgun?

**Review by dvir001 — COMMENTED** [review]
This ship has alot of security gear and gamery gear as of now.

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 9 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:5748]: also on the sprinter

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 10 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:5755]: this is on the sprinter

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 6 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:2438]: the hard worn suit you can buy from the Repdrobe and its on the sprinter ship

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 5 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:2420]: the hard worn suit is an jumpsuit that you can buy from the repdrobe that all civils have access too and its on the sprinter ship another Expedition ship

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 3 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:2359]: you can buy the sheriffs hat from an BountyVending and the trenchcoat is just renamed like it is on the sprinter

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 9 above)
Comment (dvir001) [reply Resources/Maps/Shuttles/Mushroom.yml:5748]: Getting redone with rework.

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 10 above)
Comment (dvir001) [reply Resources/Maps/Shuttles/Mushroom.yml:5755]: Getting redone with rework.

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 8 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:3820]: all expedition ships that i have send has an salvage locker but the anchor and the dart that i have seen?

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 9 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:5748]: and the gourd as an Krammar (sec gun) and the anchor has MK38s (sec guns)

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 3 above)
Comment (dvir001) [reply Resources/Maps/Shuttles/Mushroom.yml:2359]: Its getting fixed, the hat will need to get a new version with a new name..
Ill look into adding the 2 as new versions to be used for it, might as well.

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 7 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:3000]: well the sprinter has one including the dart, anchor even tho the dart and anchor isnt an merc ship

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 9 above)
Comment (dvir001) [reply Resources/Maps/Shuttles/Mushroom.yml:5748]: Ill add it to the list, we will probably end up adding new safes to just allow ships to pick from to keep some baseline

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 10 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:5755]: and i believe like 2 other ships

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 8 above)
Comment (Cheackraze) [reply Resources/Maps/Shuttles/Mushroom.yml:3820]: this is fine dw

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 3 above)
Comment (Cheackraze) [reply Resources/Maps/Shuttles/Mushroom.yml:2359]: i believe its in now with the merc gear rework

**Comment by Cheackraze** [comment]
we are indeed in the middle of reworking how weapons and such are given through ships. We are about to have 'gun lockers' in 3 tiers, corresponding to the ship's value, so it would be best not to map individual guns right now in favor of those cabinets in order to keep it in line with the rest. To be fair, however, we did determine that Kammerers are more of an advanced-civilian weapon in their current state, with the Security grade shotgun becoming the enforcer.

Also, the 'bounty hunter' variants of the items found on the sprinter, are now official spawned entities you can use again.  I super appreciate your patience here, but We've been working to add the things to make sure this ship can be everything its designed to be! Looking forward to a possible update

**Comment by VividPups** [comment]
@Cheackraze  I do apparate the Responds with the extra info about what is going on behind the scene's and what is to come, Im sure you can tell by now; The USS Mayflower was Heavily influence by the Sprinter and I wanted to make more Bounty hunter style ships. Now at this point I kinda of step away from SS13 as a whole due to some stuff that has happen on SS14 I made it clear to some people there is a good chance I wont return. However, I would still like to continue to atleast finish this ship and it seems like the best Course of action is to wait for some of thoses things to be release

**Comment by Cheackraze** [comment]
> @Cheackraze I do appreciate the responses with the extra info about what is going on behind the scenes and what is to come, I am sure you can tell by now that the USS Mayflower was Heavily influenced by the Sprinter and I wanted to make more Bounty Hunter style ships. Now at this point I kinda of stepped away from SS13 as a whole due to some stuff that has happened on SS14 I made it clear to some people that there is a good chance I won't return. However, I would still like to continue to at least finish this ship and it seems like the best Course of action is to wait for some of those things to be released

all of the mercenary gear has been updated, feel free to update the shuttle with the new items so I can merge it. Let me know or ask on discord if you need help merging the latest updates into your working branch, so that you get access to the new items.

File: Resources/Prototypes/_NF/Shipyard/mayflower.yml (hunk 1 above)
Comment (VividPups) [reply Resources/Prototypes/_NF/Shipyard/mayflower.yml:20]: what do you mean by that? Where is the space post to be at?

Comment (Cheackraze) [reply Resources/Prototypes/_NF/Shipyard/mayflower.yml:20]: > what do you mean by that? Where is the space post to be at?

between Mayflower and {1}

**Comment by Cheackraze** [comment]
so, full review on whats presented. I have 2 issues with it overall, but they are mostly minor. One is the lack of dirt. Since the theme here is abandoned/cobweb infested/broken down, I feel that adding (cleanable) dirt decals throughout the ship will really bring it all together. And second, is the use of kitty corner wall design. 

mayflower

Although its tempting to try to save space using this, it has a poor visual effect on the walls and lighting system in the game. With icon smoothing on walls, you get strange looking walls and can 'see' on the other side of the walls where it doesnt connect and looks bad. Another issue is just the fact that it creates weak points in the design, and have the potential to react funky with projectiles passing 'through' that point in the wall. Among other things.  I feel that you definitely have plenty of space inside the ship to adjust some of the rooms and shapes to avoid this issue, while still keeping true to your design goals 100%. Otherwise, I like the design and idea behind it, and this will likely join the bison and SV series shuttles (maybe a few others like kestrel, maybe pathfinder...) in a 'scrapyard' shipyard catalogue.

File: Resources/Prototypes/_NF/Shipyard/mayflower.yml (hunk 1 above)
Comment (VividPups) [reply Resources/Prototypes/_NF/Shipyard/mayflower.yml:20]: ooo ok also... i just want to say I appearnlty mess this up really big so I am right now fixing all of this with the help of minty

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 3 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:2359]: I have change the  custom bounty hunter name hos cloths To actual  Bounty hunter cloths

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 5 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:2420]: these has been corrected

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 6 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:2438]: this has been change to bounty

File: Resources/Maps/Shuttles/Mushroom.yml (hunk 7 above)
Comment (VividPups) [reply Resources/Maps/Shuttles/Mushroom.yml:3000]: Drain has been removed

**Comment by VividPups** [comment]
So pretty much what happened was. I appeared wasn't doing it right in a way where I couldn't update the local server with the new release stuff. Which in respond. I had to create a new Ship file for the game shipyard (which now when I buy from the shipyard it just freezes the game completely).  However.  I changed the Kitty Corners mark on the picture and added dirt to all the tiles. so it look more broken down and old. At this point I have no idea what is causing the crashing now. I had ask the discord about it and right now as of now they don't know why it is crashing

**Comment by VividPups** [comment]
Ok I fix it and as I try to figure out how to update all of this Have an Updated look of the ship 
Mayflower

**Comment by VividPups** [comment]
Ok, I think I did it but not sure?

**Comment by VividPups** [comment]
I did Press Push multiple times and I guess it didn't work...I'll try again after work and if there isn't any change I'll make a new. I believe with how much the change happen last night with it could be the problem.

**Comment by VividPups** [comment]
So with the Push failing so much. I am just going to close this one and then make a new one with the updated everything since for some reason this one isn't updating

**Comment by VividPups** [comment]
So I am getting an Push failed error (trying to make the new one)
... Unable to Push to the remote repositiory because your local branch is behind the remote branch. Update your branch by pulling before pushing.

but both of the pulling is gray out and im not sure how to fix that

**Comment by VividPups** [comment]
At this point im at a lost.. and don't know what to do to be able to make the new Pull request since this one isn't allowing me to update it

**Comment by VividPups** [comment]
https://github.com/new-frontiers-14/frontier-station-14/pull/734#issue-2053183009

Link to the new Request with the updates that was ask/told to do.

**Pull request closed without merging**